File: backend/DTOs/UserAdminDto.cs
using System;

namespace backend.DTOs
{
    public class UserAdminDto
    {
        public int Id { get; set; }
        public long TgUserId { get; set; }
        public string UserName { get; set; } = null!;
        public string LanguageCode { get; set; } = string.Empty;
        public int Points { get; set; }
        public int TypeUserId { get; set; }
        public string TypeUserName { get; set; } = string.Empty;
        public DateTimeOffset RegisteredAt { get; set; }
        public DateTimeOffset UpdatedAt { get; set; }

        /// <summary>False when the account has been disabled (soft deleted).</summary>
        public bool Status { get; set; }
    }
}

File: frontend/src/app/pages/desktop/users/models/user-admin-dto.ts
export interface UserAdminDto {
  readonly id: number;
  readonly tgUserId: number;
  readonly userName: string;
  readonly languageCode: string;
  /** Derived from the user's active transactions; read-only for the client. */
  readonly points: number;

  readonly typeUserId: number;
  readonly typeUserName: string;
  readonly registeredAt: string;
  readonly updatedAt: string;

  /** Soft-delete flag. `false` means the account is disabled. */
  readonly status: boolean;
}

/** Body accepted by the admin create and update endpoints. */
export interface UserPayload {
  readonly userName: string;
  readonly tgUserId: number;
  readonly languageCode: string;
  readonly typeUserId: number;
  readonly status: boolean;
}

/** Selectable account type (SuperAdmin / Admin / User). */
export interface UserRoleDto {
  readonly id: number;
  readonly name: string;
}
